src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/ObjectHeader.cs

```diff
@@ -152,7 +152,7 @@ private static unsafe int AssignHashCode(object o, int* pHeader)
                 // there is nothing - try set hashcode inline
                 Debug.Assert((oldBits & BIT_SBLK_IS_HASH_OR_SYNCBLKINDEX) == 0);
                 int newBits = BIT_SBLK_IS_HASH_OR_SYNCBLKINDEX | BIT_SBLK_IS_HASHCODE | oldBits | newHash;
-                if (Interlocked.CompareExchange(ref *pHeader, newBits, oldBits) == oldBits)
+                if (Interlocked.CompareExchange(pHeader, newBits, oldBits) == oldBits)
```

Comment:
it's unrelated to this PR, but afair, Arm64 has an ability to do bit-wise CompareExchange so technically it might speed up things here?